docs/source/notes/amp_examples.rst

```diff
@@ -0,0 +1,210 @@
+.. _amp-examples:
+
+Automatic Mixed Precision Examples
+==================================
+
+When following the examples below, you don't need to call ``.half()`` on your model(s) or data.
+In fact, you shouldn't.  Model weights should remain FP32.
+
+You also don't need to retune any hyperparameters.
+
+.. contents:: :local:
+
+.. _autocasting-examples:
+
+Autocasting Examples
+^^^^^^^^^^^^^^^^^^^^
+
+Under construction...
+
+.. _gradient-scaling-examples:
+
+Gradient Scaling Examples
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The code snippets below demonstrate recommended use of :class:`torch.cuda.amp.AmpScaler`.
+
+.. currentmodule:: torch.cuda.amp
+
+Typical use (1 loss, 1 optimizer)
+---------------------------------
+
+::
+
+    scaler = AmpScaler()
+    ...
+    for input, target in data:
+        optimizer.zero_grad()
+        output = model(input)
+        loss = loss_fn(output, target)
+        scaler.scale(loss).backward()
+        scaler.step(optimizer)
+        scaler.update()
+
+
+Gradient clipping
+-----------------
+
+Gradient clipping requires awareness that the gradients resulting from ``scaler.scale(loss).backward()`` are scaled.
+One simple way to account for the scale factor is by clipping to ``max_norm*scaler.get_scale()`` instead of ``max_norm``::
+
+    scaler = AmpScaler()
+    ...
+    for input, target in data:
+        optimizer.zero_grad()
+        output = model(input)
+        loss = loss_fn(output, target)
+        scaler.scale(loss).backward()
+
+        # Gradients are scaled, so we clip to max_norm*scale
+        torch.nn.utils.clip_grad_norm_(model.parameters(), max_norm*scaler.get_scale())
+
+        scaler.step(optimizer)
+        scaler.update()
+
+Here the *scaled* gradients are clipped.  ``scaler.step(optimizer)`` is aware that gradients
+have not yet been unscaled, and unscales them under the hood before calling ``optimizer.step()``.
+
+Gradient clipping with separate unscaling
+-----------------------------------------
+
+The specific case of clipping scaled gradients isn't so hard (all you have to do is clip to ``max_norm*scaler.get_scale()``).
```

Comment:
Cut the explanation of the case where you work with the scaled gradients directly. People can infer how to do that. Focus on your functionality.